hw/hw17/html_render.py

```diff
@@ -5,7 +5,109 @@
 
 """
 
+
 # The start of it all:
 # Fill it all in here.
 class Element(object):
-    pass
+
+    BASE_INDENT = "    "
+
+    def __init__(self, tag="", content=None, oneLineFlag=False,
+                 selfCloseFlag=False, **kwargs):
+        self.tag = tag
+        self.content = []
+        self.append(content)
+        self.oneLineFlag = oneLineFlag
+        self.selfCloseFlag = selfCloseFlag
```

Comment:
This is a very aware superclass. The other approach would be to just have these subclasses overwrite the appropriate methods of the super. One potential advantage of doing it the other way is that you'd simplify your Element class and adding new classes doesn't require you to rewrite anything in here.